pkg/docker/docker.go

```diff
@@ -472,17 +479,40 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) (err error) {
 		}
 	}
 
-	glog.V(2).Infof("Waiting for container")
-	exitCode, err := d.client.WaitContainer(container.ID)
-	wg.Wait()
-	if err != nil {
-		return err
-	}
-	glog.V(2).Infof("Container exited")
+	if opts.TargetImage {
+		exposedPorts := make([]string, 0)
+		for port := range imageMetadata.Config.ExposedPorts {
+			exposedPorts = append(exposedPorts, port.Port())
+		}
 
-	if exitCode != 0 {
-		return errors.NewContainerError(container.Name, exitCode, "")
+		glog.V(0).Infof("\n\n\n\n\nThe image %s has been started in container %s as a result of the --run=true option.  The container's stdout/stderr will be redirected to this command's glog output to help you validate its behavior.  You can also inspect the container with docker commands if you like.  If the container is set up to stay running, you will have to Ctrl-C to exit this command, which should also stop the container %s.  This particular invocation attempts to run on the exposed ports %+v \n\n\n\n\n", image, container.ID, container.ID, exposedPorts)
```

Comment:
OK, so I *think* this means in the Ports map at https://github.com/openshift/source-to-image/blob/master/Godeps/_workspace/src/github.com/fsouza/go-dockerclient/container.go#L131
as the PortMapping map above seems to be in reference to some deprecated `docker inspect` output.

Network settings is off of the Container struct:  https://github.com/openshift/source-to-image/blob/master/Godeps/_workspace/src/github.com/fsouza/go-dockerclient/container.go#L221-L235

And we get a ref to the container earlier in this method.

So yep, I'll switch it and give it a whirl.